src/library_fs.js

```diff
@@ -815,6 +815,9 @@ FS.staticInit();` +
       } catch (e) {
         throw e;
       } finally {
+        // update old node (called here to ensure consistency 
+        // across fs implementations)
+        old_node.parent = new_dir;
```

Comment:
Reading this again, I don't think we want this to run if an exception happens. That would match the previous behavior - the old places that set the parent were only reached if nothing was thrown. So let's move this to right after the finally.

For the comment, how about `update old node (we do this here to avoid each backend needing to)` - ?

Reply:
> Reading this again, I don't think we want this to run if an exception happens. Not running it in that case would match the previous behavior - the old places that set the parent were only reached if nothing was thrown. So let's move this to right after the finally. edit: unless that doesn't work in your case for some reason?

Nice _catch_ :D I agree that makes sense and everything still worked as expected for our use case, I updated in https://github.com/emscripten-core/emscripten/pull/21964/commits/631a8bfd68890fba9f01df3c1f04abb51436b15d.

> For the comment, how about update old node (we do this here to avoid each backend needing to) - ?

I think that update has a lot more clarity! I applied that change in https://github.com/emscripten-core/emscripten/pull/21964/commits/b555a458ad8e8e354ce3c95ce6f0c8c9995814c2.